waku/waku_metadata/protocol.nim

```diff
@@ -27,57 +27,51 @@ const RpcResponseMaxBytes* = 1024
 type
   WakuMetadata* = ref object of LPProtocol
     clusterId*: uint32
-    shards*: seq[uint32]
+    shards*: HashSet[uint32]
 
 proc respond(m: WakuMetadata, conn: Connection): Future[Result[void, string]] {.async, gcsafe.} =
-  try:
-    await conn.writeLP(WakuMetadataResponse(
-      clusterId: some(m.clusterId),
-      shards: m.shards
-      ).encode().buffer)
-  except CatchableError as exc:
-    return err(exc.msg)
+  let response = WakuMetadataResponse(
+    clusterId: some(m.clusterId),
+    shards: toSeq(m.shards)
+  )
+  
+  let res = catch: await conn.writeLP(response.encode().buffer)
```

Comment:
afaik `writeLP` can raise an exception, why removing the except?

Reply:
the `catch` bundle the exception if present into a `Result` with an error containing the exception message.

Reply:
i see. not sure i like using this kind of non-standard sugar. ok we already use nim-results, but in the repo we are not using this "way" of `catch`.

Reply:
Maybe we should :wink: